torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1458,6 +1469,14 @@ def _unshard(
         Postcondition: Each handle's ``FlatParameter`` 's data is the padded
         unsharded flattened parameter on the compute device.
         """
+        if (
+            self.all_gather_issue_limit is not None
```

Comment:
might be able to make this check less verbose by defaulting the limit to math.maxint or something.